source/tscript.js

```diff
@@ -2167,26 +2187,21 @@ let lib_canvas = {
 					}
 				},
 				"StereoAudio":{
-					"constructor":function(object, leftBuffer, rightBuffer, sampleRate){
+					"constructor":function(object, leftBuffer, rightBuffer, sampleRate, performTypeCheck){
 						if (! module.isDerivedFrom(leftBuffer.type, module.typeid_array)) this.error("/argument-mismatch/am-1", ["leftBuffer", "audio.StereoAudio.constructor", "array", module.displayname(leftBuffer)]);
 						if (! module.isDerivedFrom(rightBuffer.type, module.typeid_array)) this.error("/argument-mismatch/am-1", ["rightBuffer", "audio.StereoAudio.constructor", "array", module.displayname(rightBuffer)]);
-						
+
+						if(! module.isDerivedFrom(performTypeCheck.type, module.typeid_boolean)) this.error("/argument-mismatch/am-1", ["performTypeCheck", "audio.MonoAudio.constructor", "boolean", module.displayname(performTypeCheck)])	
 						if(! module.isDerivedFrom(sampleRate.type, module.typeid_integer)) this.error("/argument-mismatch/am-1", ["sampleRate", "audio.StereoAudio.constructor", "integer", module.displayname(sampleRate)])	
-						debugger;
+						
 						let buf = this.service.audioContext.createBuffer(2, leftBuffer.value.b.length, sampleRate.value.b);
```

Comment:
Please simply return here if this.service.audioContext is null or undefined. To me, the "service" interface is completely optional. E.g., if sound is not available for some reason, then the "service" is null. That may happen in particular in contexts without GUI, e.g., when automatically checking program code. Then the call should still succeed, but no JS audio object is ever created. The same applies to the play() method. Sorry, I should have noticed that earlier.